kf/nfconfig.go

```diff
@@ -942,3 +943,224 @@ func getHostInterfaces() (map[string]bool, error) {
 	}
 	return hostIfaces, nil
 }
+
+// AddProgramsOnInterface: AddProgramsOnInterface will add given ebpf programs on given interface
+func (c *NFConfigs) AddProgramsOnInterface(ifaceName, hostName string, bpfProgs *models.BPFPrograms) error {
+
+	if hostName != c.hostName {
+		errOut := fmt.Errorf("provided bpf programs do not belong to this host")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if ifaceName == "" || bpfProgs == nil {
+		errOut := fmt.Errorf("iface name or bpf programs are empty")
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	if _, ok := c.hostInterfaces[ifaceName]; !ok {
+		errOut := fmt.Errorf("%s interface name not found in the host", ifaceName)
+		log.Error().Err(errOut)
+		return errOut
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	for _, bpfProg := range bpfProgs.XDPIngress {
+		if c.IngressXDPBpfs[ifaceName] == nil {
+			if bpfProg.AdminStatus == models.Enabled {
+				c.IngressXDPBpfs[ifaceName] = list.New()
+				if err := c.VerifyAndStartXDPRootProgram(ifaceName, models.XDPIngressType); err != nil {
+					c.IngressXDPBpfs[ifaceName] = nil
+					return fmt.Errorf("failed to chain XDP BPF programs: %w", err)
+				}
+				log.Info().Msgf("Push Back and Start XDP program : %s seq_id : %d", bpfProg.Name, bpfProg.SeqID)
+				if err := c.PushBackAndStartBPF(bpfProg, ifaceName, models.XDPIngressType); err != nil {
```

Comment:
```suggestion
				if err := c.PushBackAndStartEbpfProgram(bpfProg, ifaceName, models.XDPIngressType); err != nil {
```
it doesn't start ebpf per se, it starts an ebpf program, if I understand correctly, hence the suggested rename

Reply:
I think this we can do it in a separate PR because we need to do lot of changes